src/Model/Casts/Caster.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace Mongolid\Model\Casts;
+
+use MongoDB\BSON\UTCDateTime;
+use Mongolid\Model\Casts\DateTime\BaseDateTimeCast;
+use Mongolid\Model\Casts\DateTime\DateTimeCast;
+use Mongolid\Model\Casts\DateTime\DateTimeImmutableCast;
+
+class Caster
+{
+    public static function castToDateTime(
+        string $cast,
+        mixed $value
+    )
+    {
+        if (!$value instanceof UTCDateTime) {
+            return null;
+        }
```

Comment:
Caso o valor não seja um UTCDateTime ou nulo, acredito que seja um melhor comportamento o código dar erro do que rodar errado.

```suggestion
        if (!$value instanceof UTCDateTime && !is_null($value)) {
            throw new UnexpectedValueException('Argument value should be of type UTCDateTime, ' . gettype($value) . ' is given');
        }
```

Reply:
Esse daqui eu acabei modelando uma exceção pra representar o erro e isolar a lógica pra formatar a mensagem. Aproveitei pra vaaalidar também caso o usuário passe um cast inválido no mongolid.